src/garage/envs/multi_env_wrapper.py

```diff
@@ -82,6 +86,12 @@ def __init__(self,
             if not isinstance(env, GarageEnv):
                 envs[i] = GarageEnv(env)
         super().__init__(envs[0])
+        if env_names is not None:
+            assert isinstance(env_names, list), 'env_names must be a list'
+            msg = ('env_names are not unique or there is not an env_name',
+                   'corresponding to each env in envs')
+            assert len(set(env_names)) == len(envs), msg
+        self._env_names = env_names
```

Comment:
I think it makes sense as is, personally.

Reply:
making env names a set means that each env_names won't be duplicated, because set entries are unique. Doing this allows us to verify that the number of unique names in env_names is the exact same as the number of envs that the wrapper is being instantiated with.

Reply:
Ah. I missed that you turn env_names into a set.